tests/basic/sharedv4_svc_test.go

```diff
@@ -171,11 +177,331 @@ var _ = Describe("{NFSServerFailover}", func() {
+		}
+
+		Step("scale the test-sv4-svc apps so that one pod runs on each worker node", func() {
+			testSv4Contexts = getTestSv4Contexts(contexts)
+			if len(testSv4Contexts) == 0 {
+				Skip("No test-sv4-svc apps were found")
+			}
+			numPods = len(node.GetWorkerNodes())
+			for _, ctx := range testSv4Contexts {
+				Step(fmt.Sprintf("scale up app %s to %d", ctx.App.Key, numPods), func() {
+					applicationScaleUpMap, err := Inst().S.GetScaleFactorMap(ctx)
+					Expect(err).NotTo(HaveOccurred())
+					for name := range applicationScaleUpMap {
+						applicationScaleUpMap[name] = int32(numPods)
+					}
+					err = Inst().S.ScaleApplication(ctx, applicationScaleUpMap)
+					Expect(err).NotTo(HaveOccurred())
+				})
+			}
+		})
+
+		ValidateApplications(contexts)
+
+		Step("set HA level to 2 to verify failover and failback to the same node", func() {
+			for _, ctx := range testSv4Contexts {
+				vols, err := Inst().S.GetVolumes(ctx)
+				Expect(err).NotTo(HaveOccurred())
+				Step(
+					fmt.Sprintf("set HA level to 2 on app %s's volume %v", ctx.App.Key, vols[0]),
+					func() {
+						err := Inst().V.SetReplicationFactor(vols[0], 2, nil)
+						Expect(err).NotTo(HaveOccurred())
+					})
+				Step(
+					fmt.Sprintf("validate successful update of HA level on app %s's volume %v", ctx.App.Key, vols[0]),
+					func() {
+						newRepl, err := Inst().V.GetReplicationFactor(vols[0])
+						Expect(err).NotTo(HaveOccurred())
+						Expect(newRepl).To(BeNumerically("==", 2))
+						// ValidateContext() will fail with error "volume has invalid repl value. Expected:3 Actual:2"
+						// without the line below.
+						ctx.SkipVolumeValidation = true
+					})
+			}
+		})
+
+		Step("for each context, restart volume driver on NFS server node and verify I/O", func() {
+			for _, ctx := range testSv4Contexts {
+				vols, err := Inst().S.GetVolumes(ctx)
+				Expect(err).NotTo(HaveOccurred())
+
+				apiVol, err := Inst().V.InspectVolume(vols[0].ID)
+				Expect(err).NotTo(HaveOccurred())
+
+				// verify failover and failback by repeating the steps below
+				for i := 0; i < 2; i++ {
+					var countersBefore, countersAfter map[string]appCounter
+					var attachedNodeBefore, attachedNodeAfter *node.Node
+
+					// get the attached node before the failover
+					attachedNodeBefore, err = Inst().V.GetNodeForVolume(vols[0],
+						defaultCommandTimeout, defaultCommandRetry)
+					Expect(err).NotTo(HaveOccurred())
+
+					Step(
+						fmt.Sprintf("get counter values from node %v for app %s's volume before failover",
+							attachedNodeBefore.Name, ctx.App.Key),
+						func() {
+							countersBefore = getAppCounters(apiVol, attachedNodeBefore,
+								3*time.Duration(numPods)*time.Second)
+						})
+
+					Step(
+						fmt.Sprintf("failover #%d for app %s by stopping the volume driver on node %s",
+							i, ctx.App.Key, attachedNodeBefore.Name),
+						func() {
+							StopVolDriverAndWait([]node.Node{*attachedNodeBefore})
+						})
+
```

Comment:
Don't we need to wait here so that we give time for failover to happen?

Reply:
I will add a short sleep here, just in case. But in practice, the PX takes longer to restart than the time taken for the "aggressive" failover. So, we the failover is assured even if you just to "systemctl restart portworx".